Automattic/simplenote-ios
# Log out user account after account deletion success

### Fix
Per some discussions with in app account deletion for MacOS, we decided that it would be best to log out a user once the account deletion is complete, instead of simply displaying the onboarding.  We did this on macOS to deal with some windowing bugs, which were not apparent on iOS, but could lead to being able to navigate the app after an account was deleted making it seem like the deletion had failed.

By logging the user out the local data is removed and there is no way to navigate the app. 
### Test
1.  Create an account to delete
2. delete that account.
3. open simplenote iOS back up, notice that with in a moment or two the onboarding VC will appear.
4. send simplenote to the background and open it again
5. You will see the slide out menu for a short moment (this is in part because of the restoration state and I do not think is avoidable) but it can't be interacted with and the onboarding vc should appear quickly

### Review
***(Required)*** Add instructions for reviewers.  For example:
> Only one developer are required to review these changes, but anyone can perform the review.

### Release

> These changes do not require release notes.
